lib/events.js

```diff
@@ -55,10 +55,48 @@ class EventEmitter {
   }
 
   toIterator(name) {
-    const stub = () => {};
-    this.events.on(name, stub);
+    const data = [];
+    const promises = [];
+    let error = null;
+    const handleEvent = (...value) => {
+      if (promises.length) {
+        promises.shift().resolve({ value, done: false });
+      } else {
+        data.push(value);
+      }
+    };
+    const handleError = (err) => {
+      this.off(name, handleEvent);
+      this.off('error', handleError);
+      if (err && promises.length) {
+        promises.shift().reject(err);
+      } else if (err) {
+        error = err;
+      }
+    };
+    this.on(name, handleEvent);
+    this.on('error', handleError);
     return {
-      async next() {},
+      [Symbol.asyncIterator]() {
+        return {
+          next() {
+            return new Promise((resolve, reject) => {
+              if (error) reject(error);
+              const value = data.shift();
+              if (value) resolve({ value, done: false });
+              else promises.push({ resolve, reject });
+            });
```

Comment:
- `if (error) return Promise.reject(error);`
- We need flag `isPending: boolean`